lib/src/revset.rs

```diff
@@ -439,6 +440,19 @@ impl RevsetExpression {
         Rc::new(Self::Difference(self.clone(), other.clone()))
     }
 
+    /// Commits that are in the first expression in `expressions` that is not
+    /// `none()`.
+    pub fn coalesce(expressions: &[Rc<Self>]) -> Rc<Self> {
+        match expressions {
+            [] => Self::none(),
+            [expression] => expression.clone(),
+            [expression, remainder @ ..] => Rc::new(Self::Coalesce(
+                expression.clone(),
+                Self::coalesce(remainder),
+            )),
```

Comment:
nit: If we use recursion, it might be better to split at middle so that long sequence of expressions can be processed.

To build a right-recursive tree, maybe `rfold()` can be used instead.
https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html#method.rfold

Reply:
Opted to just use recursion for now instead.